### DISCUSSION TITLE
@include is not a function adonis v5

### DISCUSSION THREAD

**Discussion by gabbmariz** [adonisjs/core#1980] [General]
Looking at the preview of adonis to see how the development of the new version is going, I came across the following error:

`ERROR: this.scopes.flat is not a function`

I analyzed the syntax of the code a lot and stopped with the **@include** of the **.edge** files and realized that the error was related to it ...

**[IMAGE](https://ibb.co/SsHdpvM)**

**Comment by gabbmariz** [comment]
This happened in a redone test, I generated the application from the new update.

This problem occurs in development, running the server with **typescript** directly, I don’t know if in the production environment you also get an error or **@include** works

**Comment by woubuc** [comment]
I don't think I'll be able to answer your question but more details about the error you got & the code that caused the error would probably help others to provide a better answer.

**Reply by gabbmariz** [reply]
> I don't think I'll be able to answer your question but more details about the error you got & the code that caused the error would probably help others to provide a better answer.

simply using **@include** in the **_.edge_** file causes the error.

**`this.scopes.flat is not a function`**

I have two pages _"home.edge"_ and _"test.edge"_ the _home_ contains **@include('test')** and an **h1** with the text _"Hello world"_ already the _test_ has an **h2** with _"hello"_.

The right thing would be for the file to be imported to the _home_, both are in the same directory, but adonis returns an error saying that **@include** is not a function.

when I remove **@include('test')** the page renders normally, with the error ...

I made two simple pages with only two html tags for testing the error can be seen in the [link ](https://ibb.co/SsHdpvM).

remembering that this error only happens when I import another page, if I don't use **@include** the page renders normally.

**Reply by woubuc** [reply]
Just checking - does your edge template contain `@import` or `@include`? Cause I'm pretty sure it should be `@include` but here you say `@import`.

**Reply by gabbmariz** [reply]
> Just checking - does your edge template contain `@import` or `@include`? Cause I'm pretty sure it should be `@include` but here you say `@import`.

sorry it's **@include**, it was a typo

**Reply by gabbmariz** [reply]
can you use @include normally in your tests?

**Comment by gabbmariz** [comment]
so to understand it is not being found a function for **@include**, **@adonisjs/view** is in version **3.0.9**
